packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1029,6 +1029,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             this.mc.logger,
             existing,
             metadata,
+            this.context.clientDetails.type === summarizerClientType,
```

Comment:
If we're not the summarizer client, do we even need to create a garbage collector? Just an optimization thought. Maybe we do.

Reply:
Ideally we shouldn't. We do need it now for the inactiveObjectLoaded events while we track down all bugs in apps. Once we do that, we should consider skipping its creation for non-summarizer clients.